packages/shared/trade/src/transaction-sidebar/transaction-sidebar.tsx

```diff
@@ -87,27 +86,28 @@ export const TransactionSidebar = ({
   const [showSwitchNetwork, setShowSwitchNetwork] = useState(false);
   const { canSubmit, confirm, tradeType, debt, collateral, selectedNetwork } =
     state;
-  const walletConnectedNetwork = useWalletConnectedNetwork();
-  const isReadyOnlyWallet = useReadOnlyAddress();
+  const {
+    wallet: { userWallet },
+  } = useAppStore();
   const isBlocked = useLeverageBlock();
   const approvalData = useTransactionApprovals(
     context,
     requiredApprovalAmount,
     variableBorrowRequired
   );
-  const mustSwitchNetwork = selectedNetwork !== walletConnectedNetwork;
+  const mustSwitchNetwork = selectedNetwork !== userWallet?.selectedChain;
```

Comment:
Can't you just keep the old hooks? Do you need to refactor it this way?

Reply:
We're using the selectedChain and isReadOnlyAddress so I thought I'd just use the userWallet instead of having two hooks

Reply:
If you want I can bring the old hooks back

Reply:
lets use the old hooks